test/plain-list/AutosuggestApp.test.js

```diff
@@ -131,6 +132,102 @@ describe('Plain list Autosuggest', () => {
     });
   });
 
+  describe('when typing and matches exist (when selectFirstSuggestion is true)', () => {
+    before(() => {
+      setSelectFirstSuggestion(true);
+    });
+
+    after(() => {
+      setSelectFirstSuggestion(false);
+    });
+
+    beforeEach(() => {
+      onSuggestionSelected.reset();
+      focusAndSetInputValue('p');
+    });
+
+    it('should focus on any suggestion', () => {
+      expectFocusedSuggestion('Perl');
+    });
+
+    it('should hide suggestions when Escape is pressed', () => {
+      clickEscape();
+      expectSuggestions([]);
+    });
+
+    it('should hide suggestions when input is blurred', () => {
+      blurInput();
+      expectSuggestions([]);
+    });
+
+    it('should show suggestions', () => {
+      expectSuggestions(['Perl', 'PHP', 'Python']);
+    });
+
+    it('should not clear the input when Escape is pressed', () => {
+      clickEscape();
+      expectInputValue('p');
+    });
+
+    it('should clear the input when Escape is pressed again', () => {
+      clickEscape();
+      clickEscape();
+      expectInputValue('');
+    });
+
+    it('should show suggestions when input is focused again', () => {
+      blurInput();
+      focusInput();
+      expectSuggestions(['Perl', 'PHP', 'Python']);
+    });
+
+    it('should focus suggestion when input is focused again', () => {
+      blurInput();
+      focusInput();
+      expectFocusedSuggestion('Perl');
+    });
+
+    it('should revert input value when Escape is pressed after Up/Down interaction', () => {
+      clickDown();
+      clickEscape();
+      expectInputValue('p');
+    });
+
+    it('should update input value when suggestion is clicked', () => {
+      clickSuggestion(1);
+      expectInputValue('PHP');
+    });
+
+    it('should focus on suggestion when mouse enters it', () => {
+      mouseEnterSuggestion(2);
+      expectFocusedSuggestion('Python');
+    });
+
+    it('should not have focused suggestions when mouse leaves a suggestion', () => {
+      mouseEnterSuggestion(2);
+      mouseLeaveSuggestion(2);
+      expectFocusedSuggestion(null);
+    });
+
+    it('should be called once with the right parameters when Enter is pressed', () => {
+      onChange.reset();
+      clickEnter();
+      expect(onSuggestionSelected).to.have.been.calledOnce;
+      expect(onSuggestionSelected).to.have.been.calledWithExactly(eventInstance, {
+        suggestion: { name: 'Perl', year: 1987 },
+        suggestionValue: 'Perl',
+        sectionIndex: null,
+        method: 'enter'
+      });
+
+      expect(onChange).to.have.been.calledOnce;
+      expect(onChange).to.be.calledWith(eventInstance, {
+        newValue: 'Perl',
+        method: 'enter'
+      });
```

Comment:
There is a special section `describe('inputProps.onChange', ...)`. Let's move this test closer to it.